packages/svelte/src/lib/components/Render.svelte

```diff
@@ -20,21 +20,18 @@
 <script lang="ts">
   import store from '../store'
 
-  export let component: InertiaComponentType
+  export let component: ComponentType
   export let props: PageProps = {}
   export let children: RenderProps[] = []
 
-  let prev = component
-  let key = new Date().getTime()
-
-  function updateKey(component: InertiaComponentType) {
-    if (prev !== component) {
-      prev = component
-      key = new Date().getTime()
+  let prevComponent: ComponentType
+  let key: number
+  $: {
+    if (prevComponent !== component) {
+      key = Date.now()
+      prevComponent = component
     }
   }
```

Comment:
Prior to the typescript merge we had

```
$: {
    if (prevComponent !== component) {
      key = Date.now()
      prevComponent = component
    }
  }
```

and with the typescript merge we were now having

```
function updateKey(component: InertiaComponentType) {
    if (prev !== component) {
      prev = component
      key = new Date().getTime()
  }
}


$: updateKey(component)
```

While it looks like to be the same, it ain't. Everything between ${...} is reactive and is able to trigger state update unlike when it's wrapped into a method